### PR TITLE
Add simple completion of GCS configuration

### PR DESCRIPTION
**Pull request by legionus**

### CODE REVIEW

**Comment by bparees** [comment]
@legionus please split your vendoring commit out to make this reviewable.

@coreydaley fyi, you should compare notes w/ what you're doing for s3 to make sure we have the same general pattern/flow.

**Comment by legionus** [comment]
@bparees Done

File: pkg/operator/bootstrap.go [hunk 1]
```diff
@@ -54,5 +62,25 @@ func (h *Handler) bootstrap() error {
...
 	}
 
+	if len(cr.Spec.HTTPSecret) == 0 {
+		var secretBytes [randomSecretSize]byte
+		if _, err := rand.Read(secretBytes[:]); err != nil {
+			return fmt.Errorf("could not generate random bytes for HTTP secret: %s", err)
+		}
+		cr.Spec.HTTPSecret = fmt.Sprintf("%x", string(secretBytes[:]))
+
+		logrus.Warn("No HTTP secret provided - generated random secret")
```
Comment (dmage) [inline pkg/operator/bootstrap.go]: How should it be provided?

Comment (legionus) [reply pkg/operator/bootstrap.go]: @dmage Thanks! Removed.

**Comment by coreydaley** [comment]
@bparees will do

File: pkg/operator/bootstrap.go [hunk 2]
```diff
@@ -54,5 +60,23 @@ func (h *Handler) bootstrap() error {
 		Replicas: 1,
 	}
 
+	if len(cr.Spec.HTTPSecret) == 0 {
+		var secretBytes [randomSecretSize]byte
+		if _, err := rand.Read(secretBytes[:]); err != nil {
+			return fmt.Errorf("could not generate random bytes for HTTP secret: %s", err)
+		}
+		cr.Spec.HTTPSecret = fmt.Sprintf("%x", string(secretBytes[:]))
```
Comment (bparees) [inline pkg/operator/bootstrap.go]: what is this used for?

File: pkg/storage/gcs/gcs.go [hunk 3]
```diff
@@ -33,5 +38,25 @@ func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
...
+		}
+
+		ctx := context.Background()
+
+		client, err := storage.NewClient(ctx)
+		if err != nil {
+			return err
+		}
+
+		err = client.Bucket(d.Config.Bucket).Create(ctx, projectID, nil)
```
Comment (bparees) [inline pkg/storage/gcs/gcs.go]: do you not need to check if the bucket already exists?

File: pkg/operator/bootstrap.go (hunk 2 above)
Comment (legionus) [reply pkg/operator/bootstrap.go]: https://docs.docker.com/registry/configuration/#http

> A random piece of data used to sign state that may be stored with the client to protect against tampering. For production environments you should generate a random piece of data using a cryptographically secure random generator. If you omit the secret, the registry will automatically generate a secret when it starts. If you are building a cluster of registries behind a load balancer, you MUST ensure the secret is the same for all registries.

Comment (bparees) [reply pkg/operator/bootstrap.go]: ok thanks

File: pkg/storage/gcs/gcs.go (hunk 3 above)
Comment (legionus) [reply pkg/storage/gcs/gcs.go]: That is why it's a simple gcs completion. I do not know how to do it better ))

Comment (bparees) [reply pkg/storage/gcs/gcs.go]: ok. So you're still investigating how to check if a bucket exists?

Comment (legionus) [reply pkg/storage/gcs/gcs.go]: @bparees Yes, I wanted to leave it as follow up. This PR is just a preview to understand how to update the configuration, when to do it, and how the function might looks like.

Comment (bparees) [reply pkg/storage/gcs/gcs.go]: that's reasonable

**Pull request merged**